File: security-oauth2/src/test/java/io/micronaut/security/oauth2/docs/github/GithubApiClient.java
package io.micronaut.security.oauth2.docs.github;

//tag::clazz[]
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Header;
import io.micronaut.http.client.annotation.Client;
import io.reactivex.Flowable;

@Header(name = "User-Agent", value = "Micronaut")
@Client("https://api.github.com")
public interface GithubApiClient {

    @Get("/user")
    Flowable<GithubUser> getUser(@Header("Authorization") String authorization);
}
//end::clazz[]

File: security-oauth2/src/test/kotlin/io/micronaut/security/oauth2/docs/github/GithubApiClient.kt
package io.micronaut.security.oauth2.docs.github

//tag::clazz[]
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Header
import io.micronaut.http.client.annotation.Client
import io.reactivex.Flowable

@Header(name = "User-Agent", value = "Micronaut")
@Client("https://api.github.com")
interface GithubApiClient {

    @Get("/user")
    fun getUser(@Header("Authorization") authorization: String): Flowable<GithubUser>
}
//end::clazz[]